### PR TITLE
Importing .env with comments and blank lines breaks env set

### PR DESCRIPTION
**Pull request by SonOfBytes**
Importing .env with comments and blank lines breaks
`convox env set <.env`

This creates a new parsing function to validate .env lines and has unit tests in env_ut_test.go.

A separate _ut_test file was use to keep it apart from the integration tests in _test

This should support various empty/whitespace only lines, and simple comment lines that begin with #

See issue https://github.com/convox/rack/issues/2008

### CODE REVIEW

**Comment by soulshake** [comment]
[Review Rack PR (Importing .env with comments and blank lines breaks env set #2043)](https://trello.com/c/dxJ8BVas/2096-review-rack-pr-importing-env-with-comments-and-blank-lines-breaks-env-set-2043)

File: cmd/convox/env_ut_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestParseEnvLine(t *testing.T) {
```
Comment (MiguelMoll) [inline cmd/convox/env_ut_test.go]: nit: This test can go with the other env tests inside `env_test.go`. Nice to keep them together in one file.

File: cmd/convox/env.go [hunk 2]
```diff
@@ -282,3 +290,25 @@ func cmdEnvUnset(c *cli.Context) error {
 
 	return nil
 }
+
+func parseEnvLine(line string) (key string, value string, err error) {
+	// Deal with empty lines
+	var m bool
+	if m = regexp.MustCompile(`^\s*$`).MatchString(line); m {
+		return
```
Comment (MiguelMoll) [inline cmd/convox/env.go]: nit: Seems the variables declared in the return statement are not used anywhere. To be consistent with the return statements, can we remove the named return variables and `return "","", nil` here? Same with line 303.

File: cmd/convox/env_test.go [hunk 3]
```diff
@@ -160,7 +160,7 @@ func TestEnvSetStdin(t *testing.T) {
 		test.Http{
 			Method: "POST",
 			Path:   "/apps/myapp/environment",
-			Body:   "foo=bar\nping=pong\n",
+			Body:   "# Comment followed by blank line\n\nfoo=bar\nping=pong\n",
```
Comment (MiguelMoll) [inline cmd/convox/env_test.go]: This line should be reverted back as `foo=bar\nping=pong\n` is what's returned by the test server.

File: cmd/convox/env.go (hunk 2 above)
Comment (SonOfBytes) [inline cmd/convox/env.go]: agreed - I liked named vars for the self documenting properties, you are right to it should match the style of the rest of the code.  I have added a 1 liner documenting description.

File: cmd/convox/env_test.go (hunk 3 above)
Comment (SonOfBytes) [inline cmd/convox/env_test.go]: I did update the fixture `manifest\fixtures\env-test.env` which I believe is what is fed to the test server - does this mean that this is coded elsewhere?

File: cmd/convox/env_ut_test.go (hunk 1 above)
Comment (SonOfBytes) [inline cmd/convox/env_ut_test.go]: done

File: cmd/convox/env_test.go (hunk 3 above)
Comment (SonOfBytes) [inline cmd/convox/env_test.go]: ok - I see this breaks travis-ci :(

**Comment by MiguelMoll** [comment]
Thanks for the changes. Also, I just remembered that there is some logic server side to handle some edge cases. If it's OK with you, I can take it from here and move your changes server side so every env set call can benefit from these changes (not just the CLI.

**Comment by SonOfBytes** [comment]
Great! If I can see the changes server side I will be able to learn from your carrying of the baton 😀

**Comment by MiguelMoll** [comment]
Moving these changes forward via #2055. Thanks again for the PR!

**Pull request closed without merging**